app/models/user.rb

```diff
@@ -963,12 +963,21 @@ def ensure_uniq_email
     raise Exceptions::UnprocessableEntity, 'Email address is already used for other user.'
   end
 
-  def validate_roles
+  def validate_ooo
+    return true if out_of_office != true
+    raise Exceptions::UnprocessableEntity, 'out of office start is required' if out_of_office_start_at.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office end is required' if out_of_office_end_at.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office end is before start' if out_of_office_start_at > out_of_office_end_at
+    raise Exceptions::UnprocessableEntity, 'out of office replacement user is required' if out_of_office_replacement_id.blank?
+    raise Exceptions::UnprocessableEntity, 'out of office no such replacement user' if !User.find_by(id: out_of_office_replacement_id)
+    true
+  end
+
+  def validate_roles(role)
     return true if !role_ids
-    role_ids.each do |role_id|
-      role = Role.lookup(id: role_id)
-      raise "Unable to find role for id #{role_id}" if !role
-      next if !role.preferences[:not]
+    role = Role.lookup(id: role.id)
```

Comment:
Why do we need to lookup a role that we already got via the parameter? This is an unnecessary query.